cubic-server/Dimension.cpp

```diff
@@ -108,31 +128,43 @@ void Dimension::loadOrGenerateChunk(int x, int z, std::shared_ptr<Player> player
 {
     std::lock_guard<std::mutex> _(_loadingChunksMutex);
     if (this->_loadingChunks.contains({x, z})) {
-        if (std::find_if(this->_loadingChunks[{x, z}].begin(), this->_loadingChunks[{x, z}].end(), [player](const std::weak_ptr<Player> current_weak_player) {
+        if (std::find_if(this->_loadingChunks[{x, z}].players.begin(), this->_loadingChunks[{x, z}].players.end(), [player](const std::weak_ptr<Player> current_weak_player) {
                 if (auto current_player = current_weak_player.lock())
                     return current_player->getId() == player->getId();
                 return false;
-            }) == this->_loadingChunks[{x, z}].end()) {
-            this->_loadingChunks[{x, z}].push_back(player);
+            }) == this->_loadingChunks[{x, z}].players.end()) {
+            this->_loadingChunks[{x, z}].players.push_back(player);
         }
         return;
     }
 
-    this->_world->getGenerationPool().addJob([this, x, z] {
-        // TODO: load chunk from disk if it exists
-        this->generateChunk(x, z);
-
-        // This send the chunk to the players that are loading it
-        std::lock_guard<std::mutex> _(_loadingChunksMutex);
-        for (auto weak_player : this->_loadingChunks[{x, z}]) {
-            if (auto player = weak_player.lock()) {
-                player->sendChunkAndLightUpdate(this->_level.getChunkColumn(x, z));
+    auto id = this->_world->getGenerationPool().addJob(
+        [this, x, z] {
+            std::lock_guard _(this->_playersMutex);
+            double current_min = 999999.0f; // TODO(huntears): Change this magic value xd;
+            for (auto player : this->getPlayers()) {
+                const auto &pos = player->getPosition();
+                const Vector3<double> chunkPos = {(double) x * 16, pos.y, (double) z * 16};
+                current_min = std::min(current_min, pos.distance(chunkPos));
             }
+            return static_cast<size_t>(std::ceil(current_min));
+        },
```

Comment:
This is good for now because we are only managing the overworld, but when we start having players in another dimension, this will need to change